src/main/fc/rc_modes.c

```diff
@@ -143,10 +148,22 @@ bool isRangeActive(uint8_t auxChannelIndex, const channelRange_t *range)
         return false;
     }
 
+    uint8_t channelIndex = auxChannelIndex + NON_AUX_CHANNEL_COUNT;
+
+    // On helicopter like aircraft, the throttle RC-channel is used 
+    // to indicate the headspeed to the FC. For this, AUX3 is mapped 
+    // to the throttle RC-channel. The headspeed is used to determine,
+    // which amount of collective pitch is needed to hover. woga65:
+#if defined(USE_VARIABLE_PITCH)
```

Comment:
This to me fells a bit like the wrong approach.

The first 4 channels can be remmaped, yet, they don't required special handling here.

Reply:
> This to me fells a bit like the wrong approach.
> 
> The first 4 channels can be remmaped, yet, they don't required special handling here.

I am not overly happy with and overly scared by my solution, this is why I am asking about your thoughts.

What happenes is, that if AUX3 (which is already occupied by COLLECTIVE) is used for mode switching, it is rerouted to the THROTTLE channel which represents the headspeed which represents the flight mode.

So basically I am using the throttle RC-channel to indicate the flight mode to the FC without wasting an additional AUX channel.

Reply:
I will try to look at the code that handles the mapping of the first 4 channels, and see if we can easily adapt it to also map cyclic.

It is probably possible to omit the cyclic channel from any mode selection, as we don't offer ch1-4 as options there either.

Reply:
Just to keep our thoughts sorted.

There are three channel orders to have in mind.
1. The channel order in the code (AERT or AERT56CG)
2. The default channel order shown to the user (AETR or AECR56TG for Futaba, TAER or CAER56TG for Spektrum)
3. The user defined mapping (user thinks he maps from AETR to whatever, but really maps from AERT to whatever)

So from the coder's perspective we talk about CH7 which is COLLECTIVE which looks like THROTTLE from the user's perspective (given the current state).
Note that the user sees T at the sventh position from the left which is CH7 from the user's perspective which might lend itself to be used to switch headspeeds / flight modes. This is why I routed CH7 (which is COLLECTIVE in the code) to the throttle channel.

Reply:
I think you are mixing C with T in your comment above. Channel 3 is always throttle on futaba. Pitch/Collective is controlled by channel 6 in both futaba and spektrum. (I fly my helis with futaba radios)

So futaba woule be:
AETRC12346
And Spektrum
TAERC12346

Channel 5 would usually be connected to the tail gyro, but is probably irrelevant to this discussion.

The main issue is that stick commands rely on knowing the stick positions, but In helis, the throttle channel is no longer operated directly by stick position.

It is operated through a mix that changes the values of throttle channel and collective channel based on stick position.

That is what is creating the issue, correct?

The proposed solution is to use the collective channel to detect stick commands, as that is more likely to be set to a linear curve.

I don't think we should use channel 7 as collective by default, though, since that is not the stablished convention.

Reply:
just to make it clearer:

-  I am ok using collective for stick commands. It just need to be documented somewhere for the users, when inevitably someone complains stick commands don't work for helis :)
- I think collective should be aux2  (channel 6)
- I am ok removing collective channel from mode selection
- I am ok with having collective configurable, but we are probably ok with it hard coded to channel 6.